**Cargamos librerías y datos**

In [7]:
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as mp

data = pd.read_csv("avistamientos_aerostatos.csv")


**Carga y exploración estructural**

In [ ]:
print("\nDimensiones del dataset")
fila_columna = data.shape
print(f"Filas -> {fila_columna[0]}")
print(f"Columnas -> {fila_columna[1]}")

print("Inicio")
display(data.head(10))
print("Fin")
display(data.tail(10))
print("\n-> Vemos que al menos los últimos 10 y los primeros 10 están repetidos")


print("\nComprobación de nulos")
print("=======================")

nulos = []
for s in data:
    if np.any(pd.isna(data[s])):
        nulos.append(data[s].name)



if nulos.__len__() <= 0:
    print("\n -- NO HAY NULOS --\n")
else:
    print("\nColumnas con nulos:")
    for i in nulos:
        print("- ",i)


print("\nTipos de datos")
print("==============")
print(data.dtypes)

print("\nOptimicemos memoria convirtiendo los tipo object en distintos tipos primarios o categorizando\n")
print("- ID_Registro -> str")
print("- Pais_Origen -> str")
print("- Tecnologia_Detectada -> str")
print("- Fecha_Avistamiento -> datetime")

data["ID_Registro"] == data["ID_Registro"].astype(str)
data["Pais_Origen"] == data["Pais_Origen"].astype(str)
data["Tecnologia_Detectada"] == data["Tecnologia_Detectada"].astype(str)
data["Fecha_Avistamiento"] == pd.to_datetime(data["Fecha_Avistamiento"])
data["Fecha_Avistamiento"] == data["Fecha_Avistamiento"]


# print("\n-> Vemos que en la columna Pais_Origen hay un celdas rellenas como Desconocido, hay que vigilar esa columna.")
print("\n-> Hay bastantes filas repetidas, todos los de la cola están también en el inicio del fichero.")
print("Registros con duplicados: ",data.ID_Registro.count())
data = data.drop_duplicates()
print("Registros con duplicados: ",data.ID_Registro.count())

print("\nVemos el resumen estadístico del dataset")
display(data.describe().round(2))




Dimensiones del dataset
	Filas -> 500
Columnas -> 9
Inicio


,ID_Registro,Pais_Origen,Altitud_km,Duracion_Vuelo_h,Velocidad_Viento_kmh,Tecnologia_Detectada,Cercania_Base_Militar_km,Fecha_Avistamiento,Indice_Maniobra
0,REG-1000,EE.UU.,150.00,76.1,82.6,Vigilancia Radar,35.40,2023-02-07,49.38
1,REG-1001,Desconocido,0.50,10.4,94.1,Inteligencia de Señales,103.66,2023-01-31,21.20
2,REG-1002,EE.UU.,25.24,5000.0,39.2,Vigilancia Radar,34.02,2023-01-31,63.00
3,REG-1003,China,29.62,23.0,12.3,Meteorológico,60.58,2023-01-02,96.11
4,REG-1004,Rusia,20.83,26.7,19.0,Meteorológico,246.80,2023-02-06,101.54
5,REG-1005,EE.UU.,20.83,41.5,116.5,Inteligencia de Señales,7.05,2023-01-17,22.01
6,REG-1006,EE.UU.,29.90,19.4,42.5,Vigilancia Radar,16.06,2023-01-08,67.12
7,REG-1007,China,25.84,33.0,94.6,Vigilancia Radar,25.09,2023-01-09,33.77
8,REG-1008,Rusia,19.65,18.0,78.7,Inteligencia de Señales,27.39,2023-02-27,32.41
9,REG-1009,China,24.71,24.7,52.0,Meteorológico,26.47,2023-01-29,69.65


Fin


,ID_Registro,Pais_Origen,Altitud_km,Duracion_Vuelo_h,Velocidad_Viento_kmh,Tecnologia_Detectada,Cercania_Base_Militar_km,Fecha_Avistamiento,Indice_Maniobra
490,REG-1490,Desconocido,21.96,95.4,98.3,Meteorológico,78.65,2023-01-06,30.80
491,REG-1491,Corea del Norte,29.40,110.5,72.9,Meteorológico,119.24,2023-02-26,47.63
492,REG-1492,Corea del Norte,22.39,56.0,66.4,Inteligencia de Señales,143.85,2023-02-05,56.86
493,REG-1493,Desconocido,17.69,22.5,42.3,Inteligencia de Señales,161.11,2023-01-29,84.70
494,REG-1494,Rusia,29.62,29.7,112.5,Inteligencia de Señales,36.85,2023-01-25,32.07
495,REG-1495,Desconocido,24.69,46.0,53.7,Meteorológico,189.29,2023-01-18,49.29
496,REG-1496,Rusia,16.81,11.1,19.6,Meteorológico,70.80,2023-01-07,92.50
497,REG-1497,China,21.05,16.8,77.9,Meteorológico,8.82,2023-01-08,58.76
498,REG-1498,EE.UU.,NaN,15.5,22.5,NaN,32.40,2023-01-07,88.12
499,REG-1499,EE.UU.,15.09,59.0,48.0,Meteorológico,63.35,2023-01-28,77.31



-> Vemos que al menos los últimos 10 y los primeros 10 están repetidos

Comprobación de nulos

Columnas con nulos:
-  Altitud_km
-  Tecnologia_Detectada

Tipos de datos
ID_Registro                  object
Pais_Origen                  object
Altitud_km                  float64
Duracion_Vuelo_h            float64
Velocidad_Viento_kmh        float64
Tecnologia_Detectada         object
Cercania_Base_Militar_km    float64
Fecha_Avistamiento           object
Indice_Maniobra             float64
dtype: object

Optimicemos memoria convirtiendo los tipo object en distintos tipos primarios o categorizando

- ID_Registro -> str
- Pais_Origen -> str
- Tecnologia_Detectada -> str
- Fecha_Avistamiento -> datetime

-> Hay bastantes filas repetidas, todos los de la cola están también en el inicio del fichero.
Registros con duplicados:  500
Registros con duplicados:  500

Vemos el resumen estadístico del dataset


,Altitud_km,Duracion_Vuelo_h,Velocidad_Viento_kmh,Cercania_Base_Militar_km,Indice_Maniobra
count,460.00,500.00,500.00,500.00,500.00
mean,22.25,58.83,65.54,50.47,54.39
std,7.76,223.93,31.50,50.57,23.24
min,0.50,1.50,10.40,0.01,3.48
25%,18.48,24.70,38.35,15.81,34.90
50%,22.09,41.65,66.95,33.50,53.79
75%,25.23,63.70,93.12,70.63,72.62
max,150.00,5000.00,119.80,306.44,116.62


**Altitud_km**  
media: 22.25  
    mediana: 22.09  
        -> Distribución bastante simétrica
    desviación: 7.76
        -> Baja dispersión de los datos
        
    - SI a tener en cuenta

**Duracion_Vuelo_h**  
media: 58.83  
mediana: 41.65
    -> Distribución bastante asimétrica - NO a tener en cuenta


In [23]:
data.Duracion_Vuelo_h.median()

41.65